src/System.Collections.Specialized/tests/NameObjectCollectionBase/SetObjectTests.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Collections.Specialized.Tests
+{
+    public class SetObjectNameObjectCollectionBaseTests
+    {
+        [Fact]
+        public void TestSetIndexValue()
+        {
+            var noc = new MyNameObjectCollection();
+            for (int i = 0; i < 10; i++)
+            {
+                var foo = new Foo();
+                noc.Add("Name_" + i, foo);
+
+                noc[i] = new Foo();
+            }
```

Comment:
I don't see any asserts here, and the name of the test isn't particularly explanatory.  What is this testing?